src/resolve.ts

```diff
@@ -0,0 +1,47 @@
+import {NonspatialScaleChannel, SCALE_CHANNELS, SPATIAL_SCALE_CHANNELS, SpatialScaleChannel, UNIT_SCALE_CHANNELS} from './channel';
+import * as log from './log';
+import {contains} from './util';
+
+export type ResolveMode = 'independent' | 'shared';
+
+export interface SpatialResolve {
+  scale: ResolveMode;
+  axis?: ResolveMode;
+}
+
+export interface NonspatialResolve {
+  scale: ResolveMode;
+  legend?: ResolveMode;
+}
+
+export type Resolve = SpatialResolve | NonspatialResolve;
+
+export function isSpatialResolve(resolve: Resolve): resolve is SpatialResolve {
```

Comment:
`axis` is optional, so it might not really be in the `resolve` you provide.  Thus, this type guard is not always correct.  (I know it might work okay for the logic below, but still can lead to confusion.)

I believe you remove these two type guard and instead check whether the channel is `NonspatialScaleChannel` or `SpatialScaleChannel`.